collections-csharp/ConcurrentQueueInCSharp/ConcurrentQueueInCSharp.Console/Consumer.cs

```diff
@@ -13,14 +13,12 @@ public Consumer(OrderMessageBus messageBus)
 
         public Task Process()
         {
-            return Task.Factory.StartNew(() =>
+            return Task.Run(() =>
             {
-                Order order;
-                string processId = Guid.NewGuid().ToString();
-
-                while (_messageBus.Fetch(out order))
+                while (_messageBus.Fetch(out var order))
                 {
-                    Console.WriteLine($"ProcessId {processId} | Processing order {order.Id}");
+                    Console.WriteLine($"ProcessId {Guid.NewGuid()} | Processing order {order.Id}");
```

Comment:
@MichalBrylKaminski - I just realized that I steered you wrong on this statement, that is my bad. I didn't look at it carefully enough. The ProcessId is something that should be constant within the executing code. So the line creating a processId, should either be put back (less the ToString, as it will be handled in the interpolation):
`var processId = Guid.NewGuid()`
Or we could just use the current executing task ID which is just a static property on the Task class: `Task.CurrentId`
https://learn.microsoft.com/en-us/dotnet/api/system.threading.tasks.task.currentid

Reply:
Indeed and I actually missed the point why I was using it in that way. Fixed.